waiter/integration/waiter/authentication_test.clj

```diff
@@ -39,17 +40,21 @@
                                                                    :permitted-user "*"
                                                                    :run-as-user (retrieve-username)
                                                                    :token token))]
-          (assert-response-status response 400)
-          (is (string/includes? body error-message)))
+          (try
+            (assert-response-status response 400)
+            (is (string/includes? body error-message))
+            (finally (delete-token-and-assert waiter-url token))))
```

Comment:
Won't these fail since the token doesn't exist. We should only delete when the token was successfully created.

Reply:
done